drivers/serial/uart_ns16550.c

```diff
@@ -1356,6 +1846,60 @@ static const struct uart_driver_api uart_ns16550_driver_api = {
 #define UART_NS16550_PCIE_IRQ_FUNC_DEFINE(n)
 #endif /* CONFIG_UART_INTERRUPT_DRIVEN */
 
+#ifdef CONFIG_UART_ASYNC_API
+#define DEV_DATA_ASYNC0(n)							       \
+	.async.tx_dma_params = {						       \
+		.dma_dev =							       \
+			DEVICE_DT_GET(DT_INST_DMAS_CTLR_BY_NAME(n, tx)),	       \
+		.dma_channel =							       \
+			DT_INST_DMAS_CELL_BY_NAME(n, tx, channel),		       \
+		.dma_cfg = {							       \
+			.source_burst_length = 1,				       \
+			.dest_burst_length = 1,					       \
+			.source_data_size = 1,                                         \
+			.dest_data_size = 1,                                           \
+			.complete_callback_en = 0,                                     \
+			.error_callback_en = 1,                                        \
+			.block_count = 1,                                              \
+			.channel_direction = MEMORY_TO_PERIPHERAL,                     \
+			.dma_slot = DT_INST_DMAS_CELL_BY_NAME(n, tx, channel),	       \
+			.dma_callback = dma_callback,				       \
+			.user_data = (void *)DEVICE_DT_INST_GET(n)		       \
+		},								       \
+	},									       \
+	.async.rx_dma_params = {						       \
+		.dma_dev =							       \
+			DEVICE_DT_GET(DT_INST_DMAS_CTLR_BY_NAME(n, rx)),	       \
+		.dma_channel =							       \
+			DT_INST_DMAS_CELL_BY_NAME(n, rx, channel),		       \
+		.dma_cfg = {							       \
+			.source_burst_length = 1,                                      \
+			.dest_burst_length = 1,                                        \
+			.source_data_size = 1,                                         \
+			.dest_data_size = 1,                                           \
+			.complete_callback_en = 0,                                     \
+			.error_callback_en = 1,                                        \
+			.block_count = 1,                                              \
+			.channel_direction = PERIPHERAL_TO_MEMORY,                     \
+			.dma_slot = DT_INST_DMAS_CELL_BY_NAME(n, rx, channel),	       \
+			.dma_callback = dma_callback,				       \
+			.user_data = (void *)DEVICE_DT_INST_GET(n)		       \
+		},								       \
+	},
+
+#define DEV_DATA_ASYNC1(n)
+#ifdef CONFIG_UART_NS16550_INTEL_LPSS_DMA
+#define DEV_DATA_ASYNC(n)							       \
+	_CONCAT(DEV_DATA_ASYNC, DT_INST_PROP(n, io_mapped))(n)
+#else
+#define DEV_DATA_ASYNC(n)							       \
+	DEV_DATA_ASYNC0(n)
+#endif /* CONFIG_UART_NS16550_INTEL_LPSS_DMA */
```

Comment:
Please do not use this kind of macro trampoline. Instead, do something similar to the `COND_CODE_*()` below.